### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by mwargan**
Updated readme install section by adding easy to follow step by step instructions. Also reorganised some sections in the install section to make more sense.

### CODE REVIEW

**Comment by converge** [comment]
@mwargan 
that's so nice! I really loved it!

I just would like to ask you, why did you remove the proxy with authentication doc?

File: README.md [hunk 1]
```diff
@@ -1929,31 +1884,6 @@ proxy_chrome_extension = create_proxy_extension(proxy)
 session = InstaPy(username=insta_username, password=insta_password, proxy_chrome_extension=proxy_chrome_extension, nogui=True)
 ```
 
-### Use a proxy (Firefox)
```
Comment (converge) [inline README.md:1932]: we added it last week

**Review by converge — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mwargan) [reply README.md:1932]: Huh - when I pulled the newest code from the Dev branch I didn't see the changes there... I was just focusing on lines 1-120

**Comment by timgrossmann** [comment]
@converge I think the removal of that proxy section is due to the state of this PR being behind of the current dev branch.
A rebase/merge to the current version will resolve this 😊 

@mwargan Thank you very much for the changes, I'll review them now.

**Comment by timgrossmann** [comment]
@mwargan I really appreciate the effort you put into this change.

Using your content as a more specific, more detailed explanation will help many people. I just think it is too "much" to look at right when going on the GitHub page. It seems like the installation process still is complex and people have to consider and read a lot before using it.

Please let me know your views on those points:
- The first view should look like the setup is simple (only `pip install instapy` should be in focus)
- The discord channel is very important, we should make it a major element
- Only installation and update should be in focus (uninstalling and specific versions are not too important)
- Instantly letting people know where they can reach out to in case of problems shows users that they will get support
- considering the technical problems several people asked @TarekJ03  about, this path description will lead to problems since people will copy paste it `cd ~/path/to/quickstart-script` 
- We should make people able to know about the parameters inside the InstaPy constructor since some people might not want to provide the credentials via the command line
- Having a basic quickstart setup to simply copy&paste can be really handy

I really like your way of describing the process of installation.
Maybe we can add a section below the table of content that will then be linked to from the initial installation "short text" right at the top of the readme, that contains what you wrote to help people go through the steps.

Sorry for the long text, I just think that since this is the number one viewed file, we have to be very considerate when designing it and making it not intimidating for new users 😊 
Let me know what you think 🙏

**Comment by mwargan** [comment]
@timgrossmann I come from a marketing background so this is fun for me :)!

I agree that it should look simple, but I find more steps better than having unclear/incomplete ones that lead to frustration. In that regard then, I’d suggest that the `pip` command auto-installs a default **quickstart** script, which would eliminate the need for step two and save space on the readme. This’ll also allow people get started and test InstaPy quicker, especially since it’s unfeasible to have a live online hosted demo.

I moved the “having an issue” section from the top and instead put the links you had to the InstaPy twitter and medium because I find its not a good idea to start with suggesting you may encounter issues :P. Instead, I suggest the readme starts with those links as it’s a kind of a demo/preview of what InstaPy is.

I’m not a user of Discord so I can’t discuss this point. All I’ll say is - where is the target market? If there are more developers than users asking questions on Discord, then I’d put more focus on where the users are asking the questions, maybe the issues section, or Reddit :P? My point is - if its very important from the development point of view, it should be emphasised in the section where contributors are likely to look, e.g. a “Contributing” section. 

Conversely, if Discord is important from a usage stand point (e.g. answering questions on ‘how do I change hashtags’ and such), then it should be prominent towards the top. I think you’ll know best which kind of people are on the Discord channel. By the way, this ties in great with what you said "Instantly letting people know where they can reach out to in case of problems shows users that they will get support" - it sure does, but both the “having issues” section and this one essentially say that, so I’d go with one or the other.

I agree that less emphasis is needed on uninstalling and on installing specific versions -these are more developer oriented use-cases.

The key thing with the installation text is it must be super clear to avoid frustration and set/match expectations: so “Now all you need is a quickstart script to start InstaPy ” would change to “a quickstart script will tell InstaPy your account details, hashtags you want to use, and any other settings you want”. That way, it’s super clear to the user what and why they are doing something. It also flips the sentence from “oh no I need to do ANOTHER thing?” to “Oh cool this is the fun part”, which also means it can be slightly longer. In the commit I submitted, the last step is on line 51, while in the previous version it was on line 65, essentially feeling like less steps all together while providing deeper context to the user.

Also "We should make people able to know about the parameters inside the InstaPy constructor since some people might not want to provide the credentials via the command line" -> already there in step 2 as "You can put in your account details now by passing the username and password parameters to the `InstaPy()` function in your **quickstart** script, like so: `InstaPy(username="abc", password="123")`, or, you can [pass them using the Command Line Interface (CLI)](#pass-arguments-by-cli)." ;).

I’d also separate the entire Docs section into a separate docs.md file, which would clean up the readme significantly. I have some charts and data from previous usage that could be interesting to put on the main readme to showcase its effectiveness (mind you I’m trying to stay safe so my charts show only 8 new followers per day ~ average). I think this is the only way to visually represent what InstaPy is doing, other than creating a GIF showing how InstaPy interacts with the browser. Either way, someone waay smarter than me once said that “a picture is worth a thousand words” so I just follow that. They never did say how much a GIF is worth :D…

So, to recap my thoughts:
1. Automate the install of a default quickstart template
2. Move the docs section to another file
3. Add some sort of examples and or visual stimulus towards the top

Anyway, time for a ☕️ :P

**Comment by timgrossmann** [comment]
@mwargan Awesome write up, thank you very much for taking the time to give me detailed explanations in your thoughts! 🙌 
I realla appreciate it 😊 

I can absolutely understand your points and actually even agree with them.

Separating the docs from the actual README definitely is one of the next steps, there already is a separate repo for all docs related (including the how to guides etc.)

Automating the installation of a quickstart file will be difficult, I always recommend people using the instapy-quickstart repository to start with since it comes with everything (templates, installation and update scripts and even run scripts to avoid the use of CLI completely)
But it's a very good point that should be discussed in the dev team 🤔 

Good point about the example and stimulus towards the start, I'm totally sold for that.

Right now, discord is like the main line for help, both technical and non technical. The issues developed from carrying everything from only actual bugs and problems, which is pretty nice. So most communication is on Discord.
You're right having the issues thing right on top might give a wrong first impression 🤔 Good to hear the marketing point of view 😋 

P.S. Sorry I missed that InstaPy arguments in your docs 😁 
Enjoy your coffee 👍

**Comment by mwargan** [comment]
@timgrossmann my pleasure! 

Side note: it could be beneficial if you'd create a place where we can all publish our plugins/add-ons to InstaPy - I'm working on a Python script to read the follower and following logs and generate and email an excel workbook, and I'd love a place to share it later on - maybe a new repository like AwesomeInstaPy?

**Comment by timgrossmann** [comment]
@mwargan Haha, thank you! 🙌 

Regarding the scripts, there is one in the instapy organisation 😋 
https://github.com/InstaPy/instapy-scripts

But I love the idea with the awesome InstaPy list 🤔 
Could be a nice place to few additional information that doesn't necessarily belong to the docs itself.

The script you're writing, is it using the exported lists you get by using the relationship tools of InstaPy?

**Comment by converge** [comment]
@mwargan maybe... we could add your idea into this project:
https://github.com/converge/instapy-dashboard

I'll start a discussion about it on Discord.

**Comment by mwargan** [comment]
@timgrossmann yup - its a global package you install, then go to the workspace logs that you want to analyse, and then type in `get-report` in the CLI. It'll read InstaPy's generated followerNum.txt and followingNum.txt to plot that data in Excel and send it via Amazon SES to a given user. Combined with CRON, that's easy monthly reporting :)

**Comment by mwargan** [comment]
@timgrossmann @converge here's a rough package: https://github.com/mwargan/mmedia-rpi3-python-scripts

**Comment by timgrossmann** [comment]
@mwargan Thank you for the explanation!

Just so you know, those files will be (in the coming months) removed from the project since they hold the same information like the database. I'm actually sorry and embarrassed to tell you that.
Those files introduce several issues into the code base and only using the DB will help get rid of them.

There is a really good tool to explore the DB, https://sqlitebrowser.org. It also provides exports to CSV for the different tables. Maybe this enables you to create an even more sophisticated script that simply uses an DB export to create your insights.

**Comment by mwargan** [comment]
@timgrossmann hehe it wouldn't be developing if I didn't have to change things as stuff gets depreciated ;) I don't think it'd be hard to switch to the SQLLite database. Are there any helpful functions already in InstaPy that I could import and use to query the database - especially maybe some functions to determine current profile and log in to the db? Edit: found some in the docs and pushed a new branch Dev with the changes

The structure and tables look great and will be very helpful! I guess I'll need to create a script to migrate old InstaPy formats (the ones without the db account following progress) from .txt to the db too...

**Comment by timgrossmann** [comment]
@mwargan Haha, true that 😉 
Good to hear you found the DB schema in the docs, the sqlite explorer is also really nice to explore the kind of data that are in your DB.

We'll do a release of the current `dev` branch tomorrow so I won't be able to incorporate your changes to the Readme and your tips until the next release but I think we'll move the actual documentation content to some specific doc repository then, maybe creating a static site in addition to also have an usable webpage that automatically displays the right state of the docs, too.

I think we won't be able to incorporate a quickstart.py file with the pip download. 
I mean we can download one with that command, but it will be located in some weird place no-one should ever touch 😅

Regarding the examples, I'll check if I can find some nice examples 😋 
Let's see.

Thanks for the script BTW. I'll take a look at it now.

**Comment by timgrossmann** [comment]
@mwargan @converge I've integrated your suggestions and changes into #4015 
Let me know what you guys think.

Do we need the steps? 🤔
